core/src/main/java/org/mule/internal/connection/ConnectionManagerAdapter.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.internal.connection;
+
+import org.mule.api.connector.ConnectionManager;
+import org.mule.api.lifecycle.Stoppable;
+import org.mule.api.retry.RetryPolicyTemplate;
+import org.mule.retry.policies.NoRetryPolicyTemplate;
+
+/**
+ * Base class for {@link ConnectionManager} implementations which contains state
```

Comment:
if the implementation is stateful or not has nothing to do with this. The javadoc should only specify that this expands the its contract with non api functionality.